### DISCUSSION TITLE
Adding new file functionality in Web Playground.

### DISCUSSION THREAD

**Discussion by vasucp1207** [codesandbox/sandpack#673] [Q&A] [answered]
Hey, I am working on a web playground using sandpack, how can I create add new file functionality in my web playground from the browser?
<img width="1432" alt="Screenshot 2022-12-25 at 12 04 02 AM" src="https://user-images.githubusercontent.com/85363195/209448810-ec32c995-b2ff-403b-9c5d-fdd8d9045547.png">

<img width="357" alt="Screenshot 2022-12-25 at 4 23 24 AM" src="https://user-images.githubusercontent.com/85363195/209452743-b0de275f-ca82-4c44-93e9-0fdf13044290.png">

In this case, a new file is created named new.js but how can I give a custom name to a file, because the new.js file can only be created once what if I have to add more files to the playground?

**Comment by danilowoz** [comment, marked as answer]
Hey, you can add multiples files using the `useSandpack` hook:

```jsx
const { sandpack } = useSandpack();

sandpack.addFile({
  "/foo.js": "",
  "/src/foo.js": ""
});
```

Hope it helps
